plugins/hydra_ray_launcher/tools/create_ami.py

```diff
@@ -0,0 +1,97 @@
+    return output
+
+
+def set_up_machine() -> None:
+    security_group_id = os.environ.get("RAY_BUILD_AMI_SEC_GROUP", "")
+    assert security_group_id != "", "Security group cannot be empty!"
+
+    # set up security group rules to allow pip install
+    _run_command(
+        f"aws ec2 authorize-security-group-egress --group-id {security_group_id} "
+        f"--ip-permissions IpProtocol=tcp,FromPort=443,ToPort=443,IpRanges=[{{CidrIp=0.0.0.0/0}}]"
+    )
+
+    yaml = "ray_cluster.yaml"
+
+    with tempfile.NamedTemporaryFile(suffix=".yaml", delete=False) as f:
+        with open(f.name, "w") as file:
+            OmegaConf.save(config=OmegaConf.load(yaml), f=file.name, resolve=True)
+        yaml = f.name
+        _run_command(f"ray up {yaml} -y")
+        _run_command(f"ray rsync_up {yaml} './setup_ami.py' '/home/ubuntu/' ")
+        _run_command(f"ray rsync_up {yaml} '../requirements.txt' '/home/ubuntu/' ")
+
+        print(
+            "Installing dependencies now, this may take a while (very likely more than 20 mins) ..."
+        )
+        _run_command(f"ray exec {yaml} 'python ./setup_ami.py' ")
+
+        # remove security group egress rules
+        _run_command(
+            f"aws ec2 revoke-security-group-egress --group-id {security_group_id} "
+            f"--ip-permissions IpProtocol=tcp,FromPort=443,ToPort=443,IpRanges=[{{CidrIp=0.0.0.0/0}}]"
+        )
+
+        # export the instance to an AMI
+        ec2_resource = boto3.resource("ec2")
+        ec2_client = boto3.client("ec2")
+        instance_id = None
+        for instance in ec2_resource.instances.all():
+            for t in instance.tags:
+                if (
+                    t["Key"] == "Name"
+                    and t["Value"] == "ray-ray_test_base_AMI-head"
+                    and instance.state["Name"] == "running"
+                ):
+                    instance_id = instance.id
+                    break
+
+        assert instance_id is not None
+
+        ami_name = (
+            f"ray_test_ami_{str(datetime.now()).replace(' ', '_').replace(':', '.')}"
+        )
+
+        ret = ec2_client.create_image(InstanceId=instance_id, Name=ami_name)
+        image_id = ret.get("ImageId")
+
+        # wait till image is ready, 30 mins for now, this could take a while
+        for i in range(60):
+            time.sleep(30)
+            image = ec2_resource.Image(image_id)
+            if image.state == "available":
+                print(
+                    f"{image} ready for use now, pls update your env variable: AWS_RAY_AMI={image}. "
+                    f"Please also update the test user's IAM policy to allow access to the new AMI."
+                )
+                break
+            else:
+                print(f"{image} current state {image.state}")
+
+        # Terminate instance now we have the AMI.
+        ec2_resource.instances.filter(InstanceIds=[instance_id]).terminate()
+
+
+if __name__ == "__main__":
```

Comment:
Why you no Hydra?!

You can use a config with ${env:VAR} to gain access to the environment variable, making this more portable and fleixble.
You call.

Reply:
:)) 

I definitely thought about Hydra-ing the script, but this doesn't really need much configuration, the security group id might be the only thing but even that is supposed to be a fixed value. 
This script is more like a bash script but written in python because it's much easier to do so :)

Reply:
actually never mind, this would be easier as a Hydra app, I will change it.

Reply:
Looking at your vaguely named config, I am not sure why you initially said that it does not need much config :).